core/src/main/scala/caliban/execution/Field.scala

```diff
@@ -46,25 +46,21 @@ case class Field(
   private[caliban] val aliasedName: String =
     if (alias.isEmpty) name else alias.get
 
-  private[caliban] lazy val allFieldsUniqueNameAndCondition: Boolean = {
-    def inner(fields: List[Field]): Boolean = {
-      val headCondition = fields.head._condition
-
-      val seen = new mutable.HashSet[String]
-      seen.add(fields.head.aliasedName)
-
-      var rem = fields.tail
-      while (rem ne Nil) {
-        val f        = rem.head
-        val continue = seen.add(f.aliasedName) && f._condition == headCondition
```

Comment:
Is there a test to verify that? I don't remember the logic well enough to know if this is correct to remove the condition 😄

Reply:
Yes there are a few tests for it. It's also not possible to get a duplication of the name here when the conditions are all the same since we already deduplicate by using `(aliasedName, condition)` as the deduplication key